### PR TITLE
Updates Result Summaries 

### PR DESCRIPTION
**Pull request by trevorb1**
### Description

Migrates from the the old results summary to new format; does not change the logic. Replaces the old node discount cost with updated result from otoole. 

~~**NOTE**: We can not merge this until otoole 1.1.4 is released.~~

### Issue Ticket Number

na

### Documentation

na

### CODE REVIEW

File: resources/otoole.yaml [hunk 1]
```diff
@@ -142,7 +142,7 @@ OperationalLifeStorage:
     indices: [REGION,STORAGE]
     type: param
     dtype: float
-    default: 10
+    default: 1
```
Comment (maartenbrinkerink) [inline resources/otoole.yaml:145]: Was this change deliberate? Doesn't matter either way as OL is defined within the workflow was just wondering

File: resources/otoole.yaml [hunk 2]
```diff
@@ -42,7 +42,7 @@ CapitalCostStorage:
     indices: [REGION,STORAGE,YEAR]
     type: param
     dtype: float
-    default: 2000
+    default: 0.001
```
Comment (maartenbrinkerink) [inline resources/otoole.yaml:45]: Was this change deliberate? Doesn't matter either way as storage costs are defined within the workflow was just wondering

File: workflow/scripts/osemosys_global/summary/constants.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+"""Constants for summary statistics"""
+
+RENEWABLES = ["GEO", "HYD", "SPV", "CSP", "WAS", "WAV", "WON", "WOF"]
```
Comment (maartenbrinkerink) [inline workflow/scripts/osemosys_global/summary/constants.py:3]: BIO should probable be included in RENEWABLES here as this is currently also the case in other constant files (e.g. powerplant/constants.py).

File: workflow/scripts/osemosys_global/summary/gen_shares.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
+"""Calcualtes Renewable Generation"""
+
+import pandas as pd
+from typing import Optional
+from constants import CLEAN, RENEWABLES, FOSSIL
+
+# do not include storage in generation share calcualtion 
+EXCLUDE = ["LDS", "SDS"]
```
Comment (maartenbrinkerink) [inline workflow/scripts/osemosys_global/summary/gen_shares.py:8]: This should be read dynamically from the storage_parameters entry in the config file since the storage technologies can change.

File: workflow/scripts/osemosys_global/summary/headline.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    return pd.DataFrame([data], columns=["Metric", "Unit", "Value"])
+
+
+def _filter_pwr_techs(production_by_technology: pd.DataFrame) -> pd.DataFrame:
+
+    df = production_by_technology.copy()
+
+    return df[
+        (df.index.get_level_values("TECHNOLOGY").str.startswith("PWR"))
+        & ~(df.index.get_level_values("TECHNOLOGY").str.startswith("PWRLDS"))
```
Comment (maartenbrinkerink) [inline workflow/scripts/osemosys_global/summary/headline.py:49]: This should be read dynamically from the storage_parameters entry in the config file since the storage technologies can change.

File: workflow/scripts/osemosys_global/summary/summarise_results.py
Comment (maartenbrinkerink) [inline workflow/scripts/osemosys_global/summary/summarise_results.py:1]: I'd link this to the constants file as well as such that the renewables_filter/fossil_filter are alligned with the entries from constants.

File: workflow/scripts/osemosys_global/summary/trade_flows.py
Comment (maartenbrinkerink) [inline workflow/scripts/osemosys_global/summary/trade_flows.py:1]: Perhaps on Monday you can walk me through how the values in TradeFlows.csv are derived. Takes me a bit too much time right now to figure that out:)

File: workflow/snakefile [hunk 6]
```diff
@@ -31,7 +32,6 @@ def get_otoole_data(otoole_config: str, var: str) -> list[str]:
         "StorageLevelYearStart",
         "StorageLevelYearFinish",
         "Trade",
-        "DiscountedOperatingCost",
```
Comment (maartenbrinkerink) [inline workflow/snakefile]: This removal caused the workflow on my end to not run (I think). I've added a commit to revert this, if that's incorrect let me know.

File: workflow/scripts/osemosys_global/summary/gen_shares.py
Comment (maartenbrinkerink) [inline workflow/scripts/osemosys_global/summary/gen_shares.py:1]: Can we also add gen_shares at the country level?

File: workflow/scripts/osemosys_global/summary/node_cost.py
Comment (maartenbrinkerink) [inline workflow/scripts/osemosys_global/summary/node_cost.py:1]: Can we also add costs at the country level (simple sum)?

File: workflow/scripts/osemosys_global/summary/capacity.py
Comment (maartenbrinkerink) [inline workflow/scripts/osemosys_global/summary/capacity.py:1]: Can we also add capacities at the country level (simple sum by tech)?

Comment (maartenbrinkerink) [reply workflow/scripts/osemosys_global/summary/capacity.py:1]: Can we also add domestic and international transmission capacities as separate entries? E.g. `FROM` and `TO` could be `CANXX`, 'USAXX' and 'CAN', 'CAN' for domestic capacities. In the current setup international transmission capacities will already be reported at nodal level, but there are often multiple nodes that import/export from/to the same international node (e.g. multiple nodes between CAN & USA).

File: workflow/scripts/osemosys_global/summary/trade_flows.py
Comment (maartenbrinkerink) [reply workflow/scripts/osemosys_global/summary/trade_flows.py:1]: Can we also add domestic and international transmission flows as separate entries? E.g. `NODE_1` and `NODE_2` could be `CANXX`, 'USAXX' and 'CAN', 'CAN' for domestic flows. Note that for domestic flows negative values should first be turned positive and summed after.

**Comment by maartenbrinkerink** [comment]
@trevorb1 This looks good! Two fixed should be made first I think (changing storage exclusions dynamically and making sure that RENEWABLE filters are in line) with some other suggestions provided (e.g. country level results).

File: resources/otoole.yaml (hunk 1 above)
Comment (trevorb1) [reply resources/otoole.yaml:145]: Was deliberate, and I just aligned it with powerplant operational life default values

File: resources/otoole.yaml (hunk 2 above)
Comment (trevorb1) [reply resources/otoole.yaml:45]: Was deliberate, and I just aligned it with powerplant capital cost default values

File: workflow/scripts/osemosys_global/summary/trade_flows.py
Comment (trevorb1) [reply workflow/scripts/osemosys_global/summary/trade_flows.py:1]: Totally! This remained unchanged from how they were calculated before (ie. it used @abhishek0208's implementation)

File: workflow/snakefile (hunk 6 above)
Comment (trevorb1) [reply workflow/snakefile]: Good catch!

File: workflow/scripts/osemosys_global/summary/summarise_results.py
Comment (trevorb1) [reply workflow/scripts/osemosys_global/summary/summarise_results.py:1]: I agree! I moved this to a new issue though, so we can do this all at once for all subdirectories. We will need to create `__init__.py` files for each directory and change import locations. So for future ref, it will be nice to have a single PR that does all these changes across many files.

File: workflow/scripts/osemosys_global/summary/carbon_intensity.py
[the remark this answers is not in the document]
Comment (trevorb1) [reply workflow/scripts/osemosys_global/summary/carbon_intensity.py:1]: Good call! Emission intensity was using production, but it should probably be using demand. I have updated the logic in this commit, but a confirmation that demand is correct would be great! Thanks! :)

File: workflow/scripts/osemosys_global/summary/capacity.py
Comment (trevorb1) [reply workflow/scripts/osemosys_global/summary/capacity.py:1]: I believe both of these points have been solved in this commit now! I will do a double check that the international trading is working correctly in the morning with a more robust example.

File: workflow/scripts/osemosys_global/summary/node_cost.py
Comment (trevorb1) [reply workflow/scripts/osemosys_global/summary/node_cost.py:1]: Add both total costs and power generation costs at country/node levels!

**Comment by trevorb1** [comment]
Thanks for the detailed feedback, @maartenbrinkerink! I believe most of your comments have now been addressed! One note, I (sorta) reverted your change of adding `DiscountedOperatingCost` to the missing otoole files list. Ended up being a typo as it should be `Operational` instead of `Operating`. Corrected in [this commit](https://github.com/OSeMOSYS/osemosys_global/pull/219/commits/fc2a0d17b022746caea98e12f0b91d0bf1cfb5d8).

File: workflow/scripts/osemosys_global/summary/carbon_intensity.py
[the remark this answers is not in the document]
Comment (maartenbrinkerink) [reply workflow/scripts/osemosys_global/summary/carbon_intensity.py:1]: I think in Slack we said that it should be the total production - production from storage and transmission. I.e. this way it would also cover the extra generation required to compensate for storage and transmission losses which would not be taken into account when using demand.

**Comment by maartenbrinkerink** [comment]
@trevorb1 I can't run the workflow at the moment. `MissingOutputException` occurs on my end for a numer of parameters (see below). 

image

For the below this can be solved by adding them to `missing` in the snakefile (not sure if that is the correct approach) but when trying to apply this to `"DiscountedCostByStorage"` the run still fails as the parameter is called in the `calculate_costs` rule but doesn't exist. I see that there is a `try/except FileExistsError` in `node_costs.py` but perhaps this is not functioning as it should?
```

"AnnualTechnologyEmissionsPenalty",
"CapitalInvestmentStorage",
"DiscountedCapitalInvestmentStorage",
"DiscountedSalvageValueStorage",  
```

**Comment by trevorb1** [comment]
Thanks for flagging those errors, @maartenbrinkerink! I have:
- Updated the emission intensity to use production with transmission and storage to be removed (pulling storage names from the config file) 
- Corrected issues resulting in failed workflows if storage/emission penalties are not included. This involved add the files you mentioned to the list of missing otoole files, with the exception of `AnnualTechnologyEmissionsPenalty`. This isnt a supported calculation (as only discounted emission penalty values are supported). That was my fault for not catching it when I merged otoole config file updates for `v1.1.4`. I have removed this from the otoole config file.

File: workflow/scripts/osemosys_global/summary/carbon_intensity.py
[the remark this answers is not in the document]
Comment (trevorb1) [reply workflow/scripts/osemosys_global/summary/carbon_intensity.py:1]: Corrected in [this commit ](https://github.com/OSeMOSYS/osemosys_global/pull/219/commits/03249193e988c49156c67420fb20a3c4b74e12c6)
